core/app/models/spree/app_configuration.rb

```diff
@@ -309,6 +309,16 @@ def shipping_rate_taxer_class
       @shipping_rate_taxer_class ||= Spree::Tax::ShippingRateTaxer
     end
 
+    # Allows providing your own class for calculating taxes on a shipping rate.
+    #
+    # @!attribute [rw] shipping_rate_tax_calculator_class
+    # @return [Class] a class with the same public interfaces as
+    #   Spree::TaxCalculator::ShippingRate
+    attr_writer :shipping_rate_tax_calculator_class
```

Comment:
I am becoming less convinced by the need for an "input"-type class, but we should absolutely discuss it at SolidusConf. I think marking it `@api private` for now is a good compromise.

WDYT @adammathys ?